plugins/orchestrator/src/components/WorkflownstancesViewerPage/ProcessDetailsViewer.tsx

```diff
@@ -0,0 +1,219 @@
+import PauseCircleFilledRoundedIcon from '@material-ui/icons/PauseCircleFilledRounded';
+import PlayCircleFilledWhiteRoundedIcon from '@material-ui/icons/PlayCircleFilledWhiteRounded';
+import PublishIcon from '@material-ui/icons/Publish';
+import RemoveCircleRoundedIcon from '@material-ui/icons/RemoveCircleRounded';
+
+import {
+  ProcessInstance,
+  ProcessInstanceState,
+} from '@janus-idp/backstage-plugin-orchestrator-common';
+
+import { Paragraph } from '../Paragraph/Paragraph';
+import ItemDescriptor from './ItemDescriptor';
+
+interface ProcessDetailsViewerProps {
+  selectedInstance: ProcessInstance | undefined;
+}
+
+const processInstanceIconCreator = (state: string) => {
+  const render = (icon: JSX.Element, text: string) => {
+    return (
+      <Paragraph>
+        <div
+          style={{
+            display: 'flex',
+            alignItems: 'center',
+            flexWrap: 'wrap',
+          }}
+        >
+          {icon}
+          <Typography
+            component="span"
+            variant="body2"
+            style={{ paddingLeft: '8px' }}
+          >
+            {text}
+          </Typography>
+        </div>
+      </Paragraph>
+    );
+  };
+
+  switch (state) {
+    case ProcessInstanceState.Active:
+      return render(<PlayCircleFilledWhiteRoundedIcon />, 'Active');
+    case ProcessInstanceState.Completed:
+      return render(<CheckCircleIcon htmlColor="#3e8635" />, 'Complete');
+    case ProcessInstanceState.Aborted:
+      return render(<RemoveCircleRoundedIcon />, 'Aborted');
+    case ProcessInstanceState.Suspended:
+      return render(<PauseCircleFilledRoundedIcon />, 'Suspended');
+    case ProcessInstanceState.Error:
+      return render(<CancelOutlinedIcon color="error" />, 'Error');
+    default:
+      return <></>;
+  }
+};
+
+const getProcessInstanceDescription = (processInstance: any) => {
+  return {
+    id: processInstance.id,
+    name: processInstance.processName,
+    description: processInstance.businessKey,
+  };
+};
+
+export const ProcessDetailsViewer = (props: ProcessDetailsViewerProps) => {
+  const { selectedInstance } = props;
+
+  const errorInfo = useMemo(() => {
+    if (!selectedInstance?.error?.message) {
+      return null;
+    }
+    const nodeName = selectedInstance.nodes.find(
+      n => n.definitionId === selectedInstance?.error?.nodeDefinitionId,
+    )?.name;
+    return (
+      <>
+        {nodeName && (
+          <>
+            <Typography variant="caption" style={{ fontWeight: 'bold' }}>
+              Node with error
+            </Typography>
+            <Paragraph>{nodeName}</Paragraph>
+          </>
+        )}
+        <Typography variant="caption" style={{ fontWeight: 'bold' }}>
+          Error message
+        </Typography>
+        <Paragraph>{selectedInstance.error.message}</Paragraph>
+      </>
+    );
+  }, [selectedInstance]);
+
+  return (
+    <InfoCard title="Details">
+      {selectedInstance === undefined && (
```

Comment:
```suggestion
      {!selectedInstance && (
```

Reply:
There is a refactoring going on and this component will likely be replaced with new code.